osu.Game.Rulesets.Osu/Objects/Drawables/SliderInputManager.cs

```diff
@@ -0,0 +1,254 @@
+            {
+                // Skip nested objects that are already judged.
+                if (nested.Judged)
+                    continue;
+
+                // Stop the process when a nested object is reached that can't be hit before the current time.
+                if (nested.HitObject.StartTime > Time.Current)
+                    break;
+
+                float radius = getFollowRadius(true);
+                double objectProgress = Math.Clamp((nested.HitObject.StartTime - slider.HitObject.StartTime) / slider.HitObject.Duration, 0, 1);
+                Vector2 objectPosition = slider.HitObject.CurvePositionAt(objectProgress);
+
+                // When the first nested object that is further outside the follow area is reached,
+                // forcefully miss all other nested objects that would otherwise be valid to be hit.
+                // This covers a case of a slider overlapping itself that requires tracking to a tick on an outer edge.
+                if (forceMiss || (objectPosition - mousePositionInSlider).LengthSquared > radius * radius)
+                {
+                    nested.MissForcefully();
+                    forceMiss = true;
+                }
+                else
+                    nested.HitForcefully();
+            }
+
+            // Enable tracking, since the mouse is within the follow area (if it were expanded).
+            updateTracking(true);
+        }
+
+        public void TryJudgeNestedObject(DrawableOsuHitObject nestedObject, double timeOffset)
+        {
+            switch (nestedObject)
+            {
+                case DrawableSliderRepeat:
+                case DrawableSliderTick:
+                    if (timeOffset < 0)
+                        return;
+
+                    break;
+
+                case DrawableSliderTail:
+                    if (timeOffset < SliderEventGenerator.TAIL_LENIENCY)
+                        return;
+
+                    // Ensure the tail can only activate after all previous ticks/repeats already have.
+                    //
+                    // This covers the edge case where the lenience may allow the tail to activate before
+                    // the last tick, changing ordering of score/combo awarding.
+                    var lastTick = slider.NestedHitObjects.LastOrDefault(o => o.HitObject is SliderTick || o.HitObject is SliderRepeat);
+                    if (lastTick?.Judged == false)
+                        return;
+
+                    break;
+
+                default:
+                    return;
+            }
+
+            if (!slider.HeadCircle.Judged)
+            {
+                if (slider.Tracking.Value)
+                {
+                    // Attempt to preserve correct ordering of judgements as best we can by forcing an un-judged head to be missed when the user has clearly skipped it.
+                    slider.HeadCircle.MissForcefully();
```

Comment:
What will happen if a player approaches the slider-head late, and still holds the key from the previous hit-object ?
Can it forcefully miss the slider-head because it detected a hit on a tick ?
The player might still have had the time to hit the head in time...
I don't know how realistic this scenario is though.

Reply:
Seems valid, although

> Or maybe a distinction can be made specifically for the case where the player didn't press any new key since the last object ? (might be better if doable)

would be more complexity than I'm willing to include.

Reply:
Disagree with this in the way that's it's proposed. It would mean someone that pixel-missed a slider would never gain tracking of the slider unless they realised in time and repressed on the follow circle.

There might be a better way of handling it, though.

Reply:
> It would mean someone that a pixel-missed slider would never gain tracking unless they realised in time and repressed either on the slider ball or slider head.
That's not fully correct, they would be able to gain tracking again once the head is missed. But the thing is they missed one element anyway (the head) so it's not that much cost to miss a few ticks along the way. And if you really want to mitigate, then there's my suggestion about key-presses. It's a bit complex but still very similar to what is done in the tracking code ?

Reply:
I applied this change in the recent commits.